pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ControllerFilePathProvider.java

```diff
@@ -66,13 +66,23 @@ private ControllerFilePathProvider(ControllerConf controllerConf)
     try {
       _dataDirURI = URIUtils.getUri(dataDir);
       LOGGER.info("Data directory: {}", _dataDirURI);
-
-      PinotFS pinotFS = PinotFSFactory.create(_dataDirURI.getScheme());
-      if (pinotFS.exists(_dataDirURI)) {
-        Preconditions
-            .checkState(pinotFS.isDirectory(_dataDirURI), "Data directory: %s must be a directory", _dataDirURI);
-      } else {
-        Preconditions.checkState(pinotFS.mkdir(_dataDirURI), "Failed to create data directory: %s", _dataDirURI);
+      try {
+          PinotFS pinotFS = PinotFSFactory.create(_dataDirURI.getScheme());
+          if (pinotFS.exists(_dataDirURI)) {
+            Preconditions
```

Comment:
We can move the `controllerConf.continueWithoutDeepStore()` check under this branch